README_eng.md

```diff
@@ -0,0 +1,299 @@
+
+     * Creating a token with corrupted data 
+       * Data field of zero length 
+       * The data field of the maximum allowed length filled with random garbage 
+       * Creation of a token with 0 issue amount 
+       * Generating a token with a long URI string 
+
+     * Creation of a token without input with MLT to pay commission 
+       * Test tx where Input with token and without MLT, output has token (without MLT) 
+       * Test tx where Input with token and without MLT, output has MLT (without token) 
+       * Test tx where Input without token but with MLT, output has MLT and token
+       * Test tx where no inputs for token 
+       * Test where less MLT at the input than you need to pay the commission 
+       * Test tx where Input and output have a token but with zero value
+  
+  2. **Testing token transfer**
+     * Standard creation of a token and sending it to a chain of persons, and from them collecting the token into one UTXO and checking that the token data has not changed, has not been burned or lost in any way.
+       * All data must be correct for this test.
+       * The token must be sent through multiple account groups.
+       * The total amount of the token must be equal to the created one.
+     * Incorrect amount of token in one input and one output
+       * The input contains the correct amount of token, but the output is incorrect
+       * In the input, the number is incorrect, but the output is correct
+       * Entry and exit with incorrect number of tokens
+     * Testing UTXO for token and return funds change
+     * Use in one MLT input to pay the commission and transfer the token at the same time 
+     * Check possibility to cause overflow. For example, let's take a few inputs where value is a maximum possible number, and one input should have the sum of these inputs values. 
+
+  3. **Testing the compatibility of the old version with the new one**
+     * Testing the compatibility of the new version with the old new
+       * Testing data encoding in a loop
+       * Testing the processing of junk and random data
+       * Testing the processing of fields that are written in a different order
+       * Testing the immutability of old versions 
+  
+  4. **Testing burning tokens**
+     * Trying to burn none-existing token
+     * Trying to burn more token value than exist in inputs
+     * Trying to burn MLT
+     * Trying to burn MLS-01
+     * Trying to burn NFT
+     * Trying to burn token without inputs for that
+     * Trying to burn existing token, but which is not in the input 
+
+**What we shall test additionally?**
+1. I can't make any limits on data fields sizes through SCALE. I'm pretty sure that Substrate checks size limits for the whole transactions because I take from framework already decoded structures. I can't see raw bytes. But I can't prove it without testing. 
+2. All maths with `u128`/`i128` should be double-checked and tested to prevent overflow and underflow.
```

Comment:
I would really prefer that we find some arbitrary precision integers to use... it's much easier to deal with, and most of the time it even serializes to smaller data since not everyone uses 16 bytes to represent their transaction. I'll look into that and see what I find.

Reply:
My biggest fear, btw, is overflow and underflow errors... people do these mistakes too often it's scary. Maybe we should write an Amount wrapper or something. Or maybe substrate did that already. We'll see.